packages/rocketchat-ui/client/views/app/directory.js

```diff
@@ -32,7 +32,7 @@ function directorySearch(config, cb) {
 				};
 			}
 
-			if (config.type === 'private') {
+			if (config.type === 'private' && RocketChat.authz.hasAllPermission('view-p-room')) {
```

Comment:
I believe it better be `view-outside-room` - either replacing `view-p-room` or with an `&&`

Reply:
adapted the `view-outside-room` permission